util/a2e/atoe_utils.c

```diff
@@ -545,22 +541,22 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
+					char *tempPattern;
 
-					b = a2e((char *)pattern, fmt - pattern);
+					tempPattern = a2e((char *)pattern, fmt - pattern);
 
 					/* Extract a double from args, this works for both doubles
 					 * and floats,
 					 * NB if we use float for a single precision floating
 					 * point number the result is wrong.
 					 */
-					len = sprintf(this.buffer, b, va_arg(args, double));
-					free(b);
-					b = e2a_string(this.buffer);
-					strcpy(this.buffer, b);
-					free(b);
-					this.buffer += len;
+					char *tempBuff = malloc(20 * sizeof(char));
```

Comment:
this is a bit concerning since I allocate an arbitrary amount of memory and hope my float or double conversion wont't take more!

Reply:
As you already said, this is concerning, having an arbitrary number here. We need to find a safer way. 

Also shouldn't we check, the return value of `sprintf`?

Reply:
the only purpose of this part is to convert a float or double number to string and the previous implementation was using the same logic but instead of a separate buffer, it was using the existing buffer. I selected 20 since it looks large enough to accommodate big numbers yet small enough to not have a big memory impact. 
The return value was used to allocate memory for conversion and I don't need it any more since I should process all the output. but I should check for success or failure. I'll add that logic!